lib/engine/corporation.rb

```diff
@@ -213,5 +222,17 @@ def transfer(ownable_type, to)
       ownables.clear
       transferred
     end
+
+    def closed?
+      @closed
+    end
+
+    def close!(game)
+      game.stock_market.close(self)
```

Comment:
you don't need to pass the game,

just do share_price.corporations.delete(self)

Reply:
ok

Added & as the share price might not have been set (in case of merging a corporation that has never been bought shares from),